src/libs/ReportUtils.ts

```diff
@@ -534,16 +534,15 @@ Onyx.connect({
     },
 });
 
-const reportActionsByReport: OnyxCollection<ReportActions> = {};
+let reportActionsByReport: OnyxCollection<ReportActions> = {};
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
-    callback: (actions, key) => {
-        if (!key || !actions) {
+    waitForCollectionCallback: true,
+    callback: (actions) => {
+        if (!actions) {
             return;
         }
-
-        const reportID = CollectionUtils.extractCollectionItemID(key);
-        reportActionsByReport[reportID] = actions;
+        reportActionsByReport = Object.fromEntries(Object.entries(actions).map(([key, value]) => [CollectionUtils.extractCollectionItemID(key as `reportActions_${string}`), value]));
```

Comment:
From looking at other places in the code that use `waitForCollectionCallback`, is this `.map()` necessary? (same comment goes for the other places)

Reply:
I believe it is necessary unless we would prefer to access reports by report id using `[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`]` instead of the `reportID`, I think it is better that we don't change too much of the functionality of this. In fact I'm going to revert some changes I made to `SidebarUtils`, so that the file is closer to its original functionality.

Reply:
```[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`]``` is the most common pattern used in the code. Looking at `IOU.ts` for example and the `allReports` object. All of the references look like:

```
iouReport = allReports?.[`${ONYXKEYS.COLLECTION.REPORT}${chatReport.iouReportID}`] ?? null;
```

I'm not able to find any `waitForCollectionCallback: true` instance that is using a `map()`. I think that's what allows for a performance gain. Since the code currently in this PR replaces `n` callbacks with a map that does `n` invocations, there really isn't any performance gain.